pkg/rancher-desktop/pages/Containers.vue

```diff
@@ -284,30 +247,9 @@ export default {
       description: '',
     });
 
-    // Info: Not sure if this can be improved, I don't like having to run it inside the `settings-read`event but I couldn't find a better way.
     ipcRenderer.on('settings-read', (event, settings) => {
       this.settings = settings;
-
-      if (!this.isNerdCtl) {
-        // INFO: We need to set ddClientReady outside of the component in the global scope so it won't re-render when we get the list.
-        containerCheckInterval = setInterval(async() => {
-          if (ddClientReady || this.containersList) {
-            return;
-          }
-
-          console.debug('Checking for containers...');
-
-          if (window.ddClient && this.isK8sReady) {
-            this.ddClient = window.ddClient;
-
-            try {
-              await this.getContainers();
-            } catch (error) {
-              console.error('There was a problem fetching containers:', { error });
-            }
-          }
-        }, 1000);
-      }
+      this.checkContainers().catch(console.error);
     });
 
     ipcRenderer.send('settings-read');
```

Comment:
This isn't related to your change, but it looks like we aren't removing the listener for `settings-read` when we destroy the containers component.

Reply:
I'm going to skip over this until later (when we transition this to using vuex).